### ISSUE TITLE
Add allowBreakingFormatting option to implement breaking format.

### ISSUE THREAD

**Issue by yukukotani** [pinterest/ktlint#656]
Add `allowBreakingFormatting` option that enables breaking formatting that might causes a compile error.

Discussion: https://github.com/pinterest/ktlint/pull/638#issuecomment-552624068

**Comment by paul-dingemans** [comment]
I do not see the value of such an option. 

The most efficient way to fix invalid enum entries is using the IntellI IDEA rename refactoring as this also fixes all usages of the enum entry. Such a change is out of scope of `ktlint` as this could require changes in other files as well.

Also when multiple invalid enum entries are found, especially in different enums, this would force the developer to commit changes all together or to revert changes before starting fixing it.

Finally the issue has almost 3 years old and did not get any traction since.
